cleanup of files for MrDoob's code style

gero3 opened the pull request:
This is a codestyle improvement PR for the src folder and the examples/js folder.

WestLangley commented:
This doesn't seem to be working in `examples/js/CurveExtras.js`, for example. Although, using http://zz85.github.io/mrdoobapproves/, that file reformats nicely.

It is largely just adding blank lines to all files, and not reformatting the code itself.

mrdoob commented:
I think we should only run this in the `src` folder. There are libraries in `examples` that we shouldn't touch.

gero3 commented:
Except the ones in 'examples/js/libs', any other file in examples/js should comply with MDCS, I think.
Or are there other exceptions  to this rule??

mrdoob commented:
```
examples/js/loaders/ctm
examples/js/loaders/gltf
examples/js/loaders/sea3d
examples/models
examples/obj
examples/scenes
```

gero3 commented:
I updated the branch to exclude the nessecary files.

WestLangley commented:
Thanks, @gero3!

gero3 commented:
Is there anything else I need to do to get this one or a similar one merged in??

WestLangley commented:
Yes. Please merge -- or redo and merge.

mrdoob commented:
Do you mind doing it again? There seem to be conflicts...

gero3 committed:
set config for codestyle

gero3 committed:
fixed files to codestyle

gero3 commented:
I fixed the conflicts now.

mrdoob commented:
I think I wouldn't touch `examples/js/ImprovedNoise.js` nor `examples/js/SimplexNoise.js`. Maybe we could move them to `js/libs`.

mrdoob commented:
`examples/js/loaders/UTF8Loader.js` is also developed outside this repo...

gero3 committed:
add new rule disallowNewlineBeforeBlockStatements

gero3 committed:
revert certain files

gero3 commented:
I added a new rule to the MDCS standard dictating that a curlybraces shouldn't be on a new line. this fixes the formatting. 

I also reverted those 3 files.

gero3 committed:
add space after comma

gero3 committed:
don't allow else statemnt on new line

gero3 commented:
I added 2 rules indicating that else isn't allowed to be on a newline and that there is a space behind a comma.

gero3 commented on examples/js/ShaderSkin.js:
```diff
@@ -338,26 +338,26 @@ THREE.ShaderSkin = {
-			"tBlur2"	: { type: "t", value: null },
-			"tBlur3"	: { type: "t", value: null },
-			"tBlur4"	: { type: "t", value: null },
+				"tBlur1"	: { type: "t", value: null },
```
Is there a reason why there is so much space here??

WestLangley replied:
They are tabs. The larger code block used to line up.

Do we have a rule that omits tabs after the 1st non-tab character?

WestLangley replied:
Or should we? : - )

gero3 replied:
The problem with this file is that even with these tabs is doens't outline correctly.

https://github.com/mrdoob/three.js/pull/6863/files#diff-a1806af1528eb3e8af7952cfb105a597R46
vs
https://github.com/mrdoob/three.js/pull/6863/files#diff-a1806af1528eb3e8af7952cfb105a597R43

the colon doesn't matchup compared to each other .

mrdoob commented:
Lets do this! 😁

mrdoob commented:
Thanks @gero3!
